src/signed-data-store/signed-data-store.ts

```diff
@@ -0,0 +1,130 @@
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  if (Number.parseInt(timestamp, 10) * 1000 > Date.now()) {
+    logger.warn(
+      `Sample is in the future, but by less than an hour, therefore storing anyway: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+  }
+
+  return true;
+};
+
+export const checkSignedDataIntegrity = (signedData: SignedData) => {
+  const { airnode, templateId, timestamp, encodedValue } = signedData;
+
+  if (!checkTimestamp(signedData)) {
+    return;
+  }
+
+  if (verifySignedData(signedData)) {
+    logger.warn(
+      `Refusing to store sample as signature does not match: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  return true;
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const setStoreDataPoint = async (signedData: SignedData) => {
+  const { airnode, templateId, signature, timestamp, encodedValue } = signedData;
+
+  if (!checkSignedDataIntegrity(signedData)) {
+    logger.warn(`Signed data received from signed data API has a signature mismatch.`);
+    logger.warn(JSON.stringify({ airnode, templateId, signature, timestamp, encodedValue }, null, 2));
+    return;
+  }
+
+  if (!signedApiStore[airnode]) {
+    signedApiStore[airnode] = {};
+  }
+
+  const existingValue = signedApiStore[airnode]![templateId];
+  if (existingValue && existingValue.timestamp >= timestamp) {
+    logger.debug('Skipping store update. The existing store value is fresher.');
+    return;
+  }
+
+  logger.debug(
+    `Storing sample for (Airnode ${airnode}) (Template ID ${templateId}) (Timestamp ${new Date(
+      Number.parseInt(timestamp, 10) * 1000
+    ).toISOString()}), ${BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8}`
+  );
+
+  signedApiStore[airnode]![templateId] = { signature, timestamp, encodedValue };
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const getStoreDataPoint = async (airnode: AirnodeAddress, templateId: TemplateId) =>
+  signedApiStore[airnode]?.[templateId];
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const clear = async () => {
+  signedApiStore = {};
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const prune = async () => {
```

Comment:
I don't see this function as useful. There is going to be a single data point for a particular beacon. Even if we have 1000 API providers (airnode addresses) with each having 1000 templates, we are storing 10^6 amount of data.

Burak has a related point about pruning data feed info (airnode address and template ID for beacons) but I similarly think that is not needed.

Reply:
Good point, I hadn't thought about that.
I agree on the second point I think: I think it's quite useful having that data present.